### PR TITLE
[onert-micro] Add core part

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr adds core part of the new onert-micro.

from draft https://github.com/Samsung/ONE/pull/12426
for issue: https://github.com/Samsung/ONE/issues/12427

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Review by SlavikMIPT — APPROVED** [review]
LGTM

**Review by Torrero — APPROVED** [review]
LGTM

File: onert-micro/onert-micro/include/core/OMRuntimeContext.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+{
+private:
+  reader::OMCircleReader _reader;
+
+public:
+  OMRuntimeContext() = default;
+  OMRuntimeContext(const OMRuntimeContext &) = delete;
+  OMRuntimeContext &operator=(const OMRuntimeContext &) = delete;
+  OMRuntimeContext &&operator=(const OMRuntimeContext &&) = delete;
+  OMRuntimeContext(OMRuntimeContext &&) = default;
```
Comment (ragmani) [inline onert-micro/onert-micro/include/core/OMRuntimeContext.h:41]: (Out of curiosity)
Move operator is deleted. but move constructor is not. Is this what you intended?

File: onert-micro/onert-micro/include/core/OMRuntimeGraph.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+{
+private:
+  OMRuntimeContext _context{};
+  OMRuntimeStorage _storage{};
+  memory::OMRuntimeAllocator _allocator{};
+
+public:
+  OMRuntimeGraph() = default;
+  OMRuntimeGraph(const OMRuntimeGraph &) = delete;
+  OMRuntimeGraph(OMRuntimeGraph &&) = default;
```
Comment (ragmani) [inline onert-micro/onert-micro/include/core/OMRuntimeGraph.h:43]: ditto

**Review by ragmani — APPROVED** [review]
LGTM (I don't know much about this PR. I just asked a question out of curiosity.)

File: onert-micro/onert-micro/include/core/OMRuntimeContext.h (hunk 1 above)
Comment (BalyshevArtem) [reply onert-micro/onert-micro/include/core/OMRuntimeContext.h:41]: Thank you for your review! Actually, yes - I need move property for `OMRuntimeContext` in one place of the code (in import part of the onert-micro) and in order to save binary I prefer delete move operator)

Comment (ragmani) [reply onert-micro/onert-micro/include/core/OMRuntimeContext.h:41]: I see. Thanks for your kindly reply.

**Pull request merged**